Source/driver/Castro.cpp

```diff
@@ -1438,7 +1439,7 @@ Castro::init (AmrLevel &old)
 {
     BL_PROFILE("Castro::init(old)");
 
-    Castro* oldlev = (Castro*) &old;
+    Castro* oldlev = dynamic_cast<Castro*>(&old);
```

Comment:
Do all the compilers we care about support this? cppreference says some implementations use RTTI

Reply:
not sure.  I can remove that.

Reply:
reverted